File: client/iih_stub/iih.ci_stub/iih.ci.ord/app/d/CiapptreatsheetAggDO.cs
using System;
using xap.mw.core.data;
using xap.mw.core.meta;

namespace iih.ci.ord.app.d
{
	/// <summary>
	/// 诊疗申请单DO
	/// </summary>
	public class CiapptreatsheetAggDO : BaseAggDO
	{	
	    private static string CIAPPTREATSHEETDO_CIAPPTREATSHEETORDO="iih.ci.ord.app.d.CiAppTreatSheetOrDO";

        public CiapptreatsheetAggDO() {
            this.setParentDO(new CiAppTreatSheetDO());
        }

  	    public CiAppTreatSheetDO getParentDO() {
		    return ((CiAppTreatSheetDO) base.ParentDO);
	    }      
	    
	    public void setParentDO(CiAppTreatSheetDO headDO) {
		    base.ParentDO=headDO;
	    }
	    
	    public CiAppTreatSheetOrDO[] getCiAppTreatSheetOrDO() {		    
		    
		    BaseDO[] dos=getChildren(Type.GetType(CIAPPTREATSHEETDO_CIAPPTREATSHEETORDO));
		    if (dos == null || dos.Length==0){return null;}
            CiAppTreatSheetOrDO[] r = new CiAppTreatSheetOrDO[dos.Length];
            for (int i = 0; i < dos.Length;i++ ){
                r[i] = dos[i] as CiAppTreatSheetOrDO;
            }
	        return r;
		    
	    }
	    
	    public void setCiAppTreatSheetOrDO(CiAppTreatSheetOrDO[] dos) {
            if (dos == null || dos.Length == 0) return;
            setChildren(dos[0].GetType(), dos);
	    }


	    public override BaseDO createParentDO()
	    {
	        return new CiAppTreatSheetDO();
	    }

        public override BaseDO createChildDO(string clzName)
	    {
            if (CIAPPTREATSHEETDO_CIAPPTREATSHEETORDO == clzName)
	        {
                return new CiAppTreatSheetOrDO();
	        }
	        return null;
	    }
	    
	    /// <summary>
        /// 获取子类全路径名数组
        /// </summary>
        /// <returns></returns>
        public override string[] getChildClassNames(){
            return  new string[]{"iih.ci.ord.app.d.CiAppTreatSheetOrDO"};
        }
	    public new static AggMeta GetMeta()
	    {
	        AggMeta meta = new AggMeta();
	        meta.OneToOneDOs = new string[]{};
	        return meta;
	    }
	    
	}
}

File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/ord/app/d/CiapptreatsheetAggDO.java
package iih.ci.ord.app.d;

import xap.sys.appfw.orm.model.agg.BaseAggDO;
import xap.mw.core.data.BaseDO;
import xap.mw.coreitf.i.IAggDesc;
import xap.mw.coreitf.i.IBaseDO;
import iih.ci.ord.app.d.desc.CiapptreatsheetAggDODesc;



/**
 * 诊疗申请单
 */
public class CiapptreatsheetAggDO extends BaseAggDO {
	private static final long serialVersionUID = 1L;

	@Override
	public CiAppTreatSheetDO getParentDO() {
		return ((CiAppTreatSheetDO) super.getParentDO());
	}

	public void setParentDO(CiAppTreatSheetDO headDO) {
		setParent(headDO);
	}

	public CiAppTreatSheetOrDO[] getCiAppTreatSheetOrDO() {
		IBaseDO[] dos = getChildren(CiAppTreatSheetOrDO.class);
		if(dos==null) return null;
		CiAppTreatSheetOrDO[] result=new CiAppTreatSheetOrDO[dos.length];
		for(int i=0;i<dos.length;i++){
			result[i]=(CiAppTreatSheetOrDO)dos[i];
		}
		return result;
	}
	
	public void setCiAppTreatSheetOrDO(CiAppTreatSheetOrDO[] dos) {
		setChildren(CiAppTreatSheetOrDO.class, dos);
	}

	@Override
	public IAggDesc getAggDesc() {
		CiapptreatsheetAggDODesc desc = new CiapptreatsheetAggDODesc();

		return desc;
	}
	
	 public  BaseDO createParentDO() {
         return new CiAppTreatSheetDO();
     }

     public  BaseDO createChildDO(String clzName) {
	     if (clzName.equals("iih.ci.ord.app.d.CiAppTreatSheetOrDO")) {
             return new CiAppTreatSheetOrDO();
         }
         return null;
     }
}